public/app/features/datasources/state/navModel.ts

```diff
@@ -53,14 +54,19 @@ export function buildNavModel(dataSource: DataSourceSettings, plugin: GenericDat
 
   const isLoadingNav = dataSource.type === loadingDSType;
 
-  const dsPermissions = {
+  const dsPermissions: NavModelItem = {
     active: false,
     icon: 'lock',
     id: `datasource-permissions-${dataSource.uid}`,
     text: 'Permissions',
     url: `datasources/edit/${dataSource.uid}/permissions`,
   };
 
+  if (highlightTrial()) {
```

Comment:
```suggestion
  if (highlightTrial() && !isLoadingNav) {
```
This avoid to add the ProBadge too soon and send a lot of RS events.